tests/decomposition/test_collect_resource_ops.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the CollectResourceOps interpreter."""
+
+# pylint: disable=wrong-import-position,wrong-import-order
+
+import pytest
+
+import pennylane as qml
+
+pytestmark = [pytest.mark.jax, pytest.mark.usefixtures("enable_disable_plxpr")]
+
+jax = pytest.importorskip("jax")
+
+from pennylane.decomposition.collect_resource_ops import CollectResourceOps
+
+
+class TestCollectResourceOps:
+    """Unit tests for the CollectResourceOps interpreter."""
+
+    @pytest.mark.unit
+    def test_flat_fn(self):
```

Comment:
I initially thought this name meant that the test was for non-pytree function arguments lol. Very nitpicky, optional suggestion:
```suggestion
    def test_flat_body_fn(self):
```